syft/generic/frameworks/hook/hook.py

```diff
@@ -271,6 +273,73 @@ def _hook_multi_pointer_tensor_methods(self, tensor_type):
                 new_method = self._get_hooked_multi_pointer_method(attr)
                 setattr(MultiPointerTensor, attr, new_method)
 
+    def _hook_string_methods(self, owner):
+
+        # Set the default owner
+        setattr(String, "owner", owner)
+
+        # Set of methods to hook/override for 'str' to String
+        attr_include = set(
+            [
+                "__add__",
+                "__eq__",
+                "__le__",
+                "__ge__",
+                "__gt__",
+                "__lt__",
+                "__ne__",
+                "__len__",
+                "__getitem__",
+                "__str__",
+                "__repr__",
+                "__format__",
+                "lower",
+                "upper",
+            ]
+        )
+
+        for attr in dir(str):
+
+            if attr in attr_include:
+
+                # Create the hooked method
+                new_method = self._get_hooked_string_method(attr)
+
+                # Add the hooked method
+                setattr(String, attr, new_method)
+
+    def _hook_string_pointer_methods(self):
+
+        # Set of methods to hook/override for 'str' to String
+        attr_include = set(
```

Comment:
this list is duplicate with the other one, try to only define it in one place for consistency purpose

Reply:
I added this list as a class set variable of `String`, so we can get the list by typing `String.methods_to_hook`